src/aind_data_transfer_service/models.py

```
@@ -200,3 +204,39 @@ def from_airflow_task_instance(
             duration=airflow_task_instance.duration,
             comment=airflow_task_instance.note,
         )
+
+
+class JobParamConfigs(BaseModel):
+    """Configs for job_type parameters in AWS Parameter Store"""
+
+    _PARAM_PREFIX: ClassVar[str] = os.getenv("AIND_AIRFLOW_PARAM_PREFIX", "")
+    _PARAM_REGEX: ClassVar[Pattern] = re.compile(
+        f"{_PARAM_PREFIX}/(?P<job_type>[^/]+)/tasks/(?P<task_id>[^/]+)"
+    )
+
+    @staticmethod
+    def get_parameter_name(job_type: str, task_id: str) -> str:
+        """Create the parameter name from job_type and task_id"""
+        return f"{JobParamConfigs._PARAM_PREFIX}/{job_type}/tasks/{task_id}"
+
+
+class JobParamInfo(BaseModel):
+    """Model for job parameter info from AWS Parameter Store"""
+
+    name: Optional[str]
+    last_modified: Optional[datetime]
+    job_type: Optional[str]
+    task_id: Optional[str]
+
+    @classmethod
+    def from_aws_describe_parameter(cls, parameter: ParameterMetadataTypeDef):
+        """Map the parameter to the model"""
+        param_name = parameter.get("Name")
+        if match := JobParamConfigs._PARAM_REGEX.match(param_name):
+            return cls(
+                name=param_name,
+                last_modified=parameter.get("LastModifiedDate"),
+                job_type=match.group("job_type"),
+                task_id=match.group("task_id"),
+            )
+        return None
```

It might be better to raise an error or return a default class like cls().

---

I wanted to return None since unexpected params can simply be ignored. This is logged in the server `list_parameters` logic.

---

You should keep the class constructor simple and move the check outside. I'm not sure what an unexpected parameter is supposed to mean in this context.

---

Moved the check outside.